tests/gateway/test-nodeport.yaml

```diff
@@ -0,0 +1,19 @@
+config:
+  serverConfig:
+    http-server.http.enabled: true
+    http-server.http.port: 8080
+    http-server.https.enabled: true
+    http-server.https.port: 8443
+    http-server.https.keystore.path: /etc/scratch/tls.pem
+
+service:
+  type: NodePort
+  ports:
+    - protocol: TCP
+      name: request
+      nodePort: 30443
```

Comment:
Note that we don't verify if this node port is reachable. We could add an extra wget in the `test-connection` in a conditional block that'd only run if the service was of type `NodePort`.